internal/oci/runtime_oci.go

```diff
@@ -784,151 +784,113 @@ func (r *runtimeOCI) UpdateContainer(ctx context.Context, c *Container, res *rsp
 	return nil
 }
 
-func WaitContainerStop(ctx context.Context, c *Container, timeout time.Duration, ignoreKill bool) error {
+// StopContainer stops a container. Timeout is given in seconds.
+func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout int64) (retErr error) {
 	ctx, span := log.StartSpan(ctx)
 	defer span.End()
 
+	if c.Spoofed() {
+		c.state.Status = ContainerStateStopped
+		c.state.Finished = time.Now()
+		return nil
+	}
+
+	if err := c.ShouldBeStopped(); err != nil {
+		if err == ErrContainerStopped {
+			err = nil
+		}
+		return err
+	}
+
+	// The initial container process either doesn't exist, or isn't ours.
+	if err := c.IsAlive(); err != nil {
+		c.state.Finished = time.Now()
```

Comment:
Do we need to set `c.state.Status` here?

Reply:
the state machine is kind of weird. TBH I don't think we should set either ever, and rely on UpdateContainerStatus to set them, but this is here for now.